export XPUStream to doc

guangyey opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121398

# Motivation
We would like to export XPUStream to public [doc](https://pytorch.org/cppdocs/api/library_root.html). The detailed documentation can help users understand and utilize XPU more effectively.

# Additional Context
A detailed XPUStream API and usage should be documented to public doc, like cuda's [doc](https://github.com/pytorch/pytorch/blob/main/docs/cpp/source/notes/tensor_cuda_stream.rst).


cc @svekars @brycebortree @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

guangyey committed:
export XPUStream to doc

[ghstack-poisoned]

albanD reviewed (commented):
That sounds fine. I would say that we have very very few users of our c++ API.
I don't see this on the rendered doc: https://docs-preview.pytorch.org/pytorch/pytorch/121398/cppdocs/search.html?q=xpustream&check_keywords=yes&area=default  (you can find this link in the drci comment)

gujinghui commented:
> That sounds fine. I would say that we have very very few users of our c++ API. I don't see this on the rendered doc: https://docs-preview.pytorch.org/pytorch/pytorch/121398/cppdocs/search.html?q=xpustream&check_keywords=yes&area=default (you can find this link in the drci comment)

@albanD  PyTorch has some C++ users, perhaps they are not very active in online community.  :)

jgong5 commented on c10/xpu/XPUStream.h:
```diff
@@ -30,21 +30,19 @@ static constexpr int max_compile_time_stream_priorities = 2;
- * a SYCL queue. On each device, a SYCL queue pool consists of kStreamsPerPool
- * queues, and you can access a particular queue by its index. The index is
- * extracted from XPUStream.id().
+ * a SYCL queue, which allows to asynchronously execute tasks on XPU.
```
```suggestion
 * a SYCL queue, which allows asynchronous execution of XPU tasks.
```

guangyey replied:
Thanks, done.

jgong5 reviewed (approved):
some nits, otherwise LGTM.

guangyey committed:
Update on "export XPUStream to doc"


# Motivation
We would like to export XPUStream to public [doc](https://pytorch.org/cppdocs/api/library_root.html). The detailed documentation can help users understand and utilize XPU more effectively.

# Additional Context
A detailed XPUStream API and usage should be documented to public doc, like cuda's [doc](https://github.com/pytorch/pytorch/blob/main/docs/cpp/source/notes/tensor_cuda_stream.rst).


cc svekars brycebortree jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

albanD commented:
cc @svekars do you know why the Library API does not render on the preview? How do we fix that?

guangyey commented:
@svekars Could you please help review this RP?

guangyey commented:
@svekars Could you help review this PR?

guangyey commented:
Hi @svekars and @albanD , what can I do for this PR?

guangyey commented:
@svekars it would be appreciated if you could help review this PR.

guangyey commented:
@svekars may I know if you have comments on this PR?

albanD reviewed (approved):
Stamping to unblock. Will ping @svekars directly if she has other comments.

guangyey commented:
> Stamping to unblock. Will ping @svekars directly if she has other comments.

Thanks for your help!

guangyey committed:
Update on "export XPUStream to doc"


# Motivation
We would like to export XPUStream to public [doc](https://pytorch.org/cppdocs/api/library_root.html). The detailed documentation can help users understand and utilize XPU more effectively.

# Additional Context
A detailed XPUStream API and usage should be documented to public doc, like cuda's [doc](https://github.com/pytorch/pytorch/blob/main/docs/cpp/source/notes/tensor_cuda_stream.rst).


cc svekars brycebortree jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

guangyey commented:
@pytorchbot rebase

guangyey committed:
Update

[ghstack-poisoned]

guangyey commented:
@pytorchbot merge
